### PR TITLE
Add min_p (rebased ig)

### PR DESCRIPTION
**Pull request by menhguin**
Initial add of min_p to _sample (the first one). I'm not used to jax so being very careful, feel free to reject and just copy paste urself.

Fairly minimal changes from https://github.com/xjdr-alt/entropix/pull/2, although I did rename
`top_p = jnp.max(probs, axis=-1, keepdims=True)`
to
`p_max = jnp.max(probs, axis=-1, keepdims=True)`

bc
1. I'm used to calling it p_max
2. might prevent future hilarious catastrophes from naming something _inside_ min_p "top_p"
3. It was a little confusing for me to read at first, I though you were doing some hybrid.

Questions:

1. **min_p for _sample high entropy, high varentropy** - it appears xdjr is planning to use for _High Entropy, High Varentropy: "resampling in the mist"_ as well. I suppose I'll just sleep and let him make his planned implementation for the Entropy Sampler phase, since he's actively making changes. Unless he wants me to take a swing at it when I wake up.
2. **Order prolly matters here** - Technically Top K you can put anywhere since the top k set will never change w before and after truncation, but putting Top P and Min P in different orders will lead to different results. It's not common to use both in normal situations (and there's no advantage I know of), IDK what shrek has planned tho. Top P usually makes sense if u wanna more deterministic and Min P is like 5% worse for low temp argmaxxing but much, much better at diversity/high temp.
3. **Unsure which is better for -inf** - I could do
`logit = jnp.where(indices_to_remove, jnp.full_like(logit, float('-inf')), logit)`
or
`logit = jnp.where(indices_to_remove, -jnp.inf, logit)` - went with the original PR and shrek can veto if he wants

will do torch implementation tmr, haven't looked over it yet. Will make more extensive PRs when I figure out what's going on/what bro is trying to do.

### CODE REVIEW

**Comment by xjdr-alt** [comment]
Good stuff. Incorporated into https://github.com/xjdr-alt/entropix/pull/11
